MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -1009,6 +993,28 @@ public Mission getMission(int id) {
         return missions.get(id);
     }
 
+    /**
+     * Add scenario to an existing mission. This method will also assign the scenario an id and place it in the scenarioId
+     * hash
+     *
+     * @param s - the Scenario to add
+     * @param m - the mission to add the new scenario to
+     */
+    public void addScenario(Scenario s, Mission m) {
+        int id;
+        if (s.getId() == Scenario.S_DEFAULT_ID) {
+            id = lastScenarioId + 1;
+            lastScenarioId = id;
+            s.setId(id);
+        } else {
+            // Scenario has already been assigned an Id, so just use its assigned value
+            id = s.getId();
```

Comment:
How does this happen?

Reply:
Removing and replacing scenarios in GM Generation forced this. I wanted to avoid it  but couldn't figure out a better way.

Reply:
I'll add a comment for that when I get home